08-5-ex-sol-ADHD.Rmd

```diff
@@ -0,0 +1,166 @@
+---
+output:
+  html_document: default
+  pdf_document: default
+---
+ * **Importing and processing data**
+
+```{r, message=FALSE, warning=FALSE}
+# Defining file paths
+biom_file_path <- "data/Aggregated_humanization2.biom"
+sample_meta_file_path <- "data/Mapping_file_ADHD_aggregated.csv"
+tree_file_path <- "data/Data_humanization_phylo_aggregation.tre"
+library(mia)
+# Imports the data
+se <- loadFromBiom(biom_file_path)
+names(rowData(se)) <- c("Kingdom", "Phylum", "Class", "Order", 
+                        "Family", "Genus")
+# Goes through the whole DataFrame. Removes '.*[kpcofg]__' from strings, where [kpcofg] 
+# is any character from listed ones, and .* any character.
+rowdata_modified <- BiocParallel::bplapply(rowData(se), 
+                                           FUN = stringr::str_remove, 
+                                           pattern = '.*[kpcofg]__')
+# Genus level has additional '\"', so let's delete that also
+rowdata_modified <- BiocParallel::bplapply(rowdata_modified, 
+                                           FUN = stringr::str_remove, 
+                                           pattern = '\"')
+# rowdata_modified is a list, so it is converted back to DataFrame format. 
+rowdata_modified <- DataFrame(rowdata_modified)
+# And then assigned back to the SE object
+rowData(se) <- rowdata_modified
+# We use this to check what type of data it is
+# read.table(sample_meta_file_path)
+# It seems like a comma separated file and it does not include headers
+# Let us read it and then convert from data.frame to DataFrame
+# (required for our purposes)
+sample_meta <- DataFrame(read.table(sample_meta_file_path, sep = ",", header = FALSE))
+# Add sample names to rownames
+rownames(sample_meta) <- sample_meta[,1]
+# Delete column that included sample names
+sample_meta[,1] <- NULL
+# We can add headers
+colnames(sample_meta) <- c("patient_status", "cohort", "patient_status_vs_cohort", "sample_name")
+# Then it can be added to colData
+colData(se) <- sample_meta
+# Convert to tse format
+tse <- as(se, "TreeSummarizedExperiment")
+# Reads the tree file
+tree <- ape::read.tree(tree_file_path)
+# Add tree to rowTree
+rowTree(tse) <- tree
+```
+
+
+ * Visualize community variation with PCoA.
+ 
+```{r, message=FALSE, warning=FALSE}
+# Computing and storing relative abundance
+tse <- transformCounts(tse, method = "relabundance")
+
+# Gets relative bundance table
+relabundance_assay <- assays(tse)$relabundance
+
+# Transposes it to get taxa to columns
+relabundance_assay <- t(relabundance_assay)
+
+# Calculates Euclidean distances between samples.
+euclidean_dist_relA <- vegan::vegdist(relabundance_assay, method = "euclidean")
```

Comment:
Euclid is typically used with CLR, not relababundance. 

Can we change this to CLR + Euclidean? Or relabundance + Bray-Curtis? Or both. These are more standard. Hmm indeed it seems we have those below, so I think relabundance + euclid case can be just removed as we do not see this in the literature in this field.

If there was something else in the exercise task, that can be changed accordingly.

Reply:
I use euclidean with counts?

Reply:
or some other distance? just for having an answer to the task of the exercise

Reply:
How about just keeping two options that are also being used in reality: 1) relabundance + bray-curtis; 2) clr + euclid (ie Aitchison distance)?

Reply:
Sure, so I just drop the " Visualize community variation with PCoA." question?

Reply:
You can visualize community variation with PCoA also by using 1) relabundance + bray-curtis and/or 2) clr + euclid (ie Aitchison distance). Do you mean that it would be good to have here something different than in the beta diversity section that already contains those examples? 

If so, that is a good point and I wrote these questions before we actually had added those examples. Sorry for the hassle.

Now, we have them already in the material, so you could change the question itself (and modify your solutions accordingly): let us replace question 1 with the following: 

"Visualize community variation with different methods (PCA, MDS, t-SNE...) by using the options in the alternative method, plotReducedDim [here link to OMA 6.2.2]. Compare results obtained with different dissimilarities (Euclidean, Bray-Curtis, Unifrac..) and transformations (CLR, compositional..) of your own choice." -> you do not have to run all these, just prepare two cases for instance, and plot them next to each other to facilitate comparison. 

In order to keep this comparable with the training material, how about just plotting the options 1-2 here as well, but this time using the function plotReducedDim as in the OMA examples? The idea is then that the students will be able to see alternative ways to do the same thing?


We could also add a new question: 4) perform community typing for the data using the DMM method (LINK to OMA community typing section)

Would this work?

Reply:
Ok I'll do that